dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -0,0 +1,751 @@
+
+        def update():
+            status.value = self._widget_status()
+
+        pc = PeriodicCallback(update, 500, io_loop=self.loop)
+        self.periodic_callbacks["cluster-repr"] = pc
+        pc.start()
+
+        return box
+
+    def scale(self, workers=1):
+        """ Scale the cluster. We can add or reduce the number workers
+        of a given configuration
+
+        Example
+        ----------
+        ```python
+        cluster.scale(2)
+        ```
+        """
+        if workers <= 0:
+            self.close()
+            return
+
+        count = len(self.workers_list) + 1  # one more worker in head node
+
+        if count < workers:
+            self.scale_up(workers - count)
+        elif count > workers:
+            self.scale_down(count - workers)
+        else:
+            print(f"Number of workers: {workers}")
+
+    # scale up
+    def scale_up(self, workers=1):
+        run_config = RunConfiguration()
+        run_config.target = self.compute_target
+        run_config.environment = self.environment_definition
+
+        scheduler_ip = self.run.get_metrics()["scheduler"]
+        args = [
+            f"--scheduler_ip_port={scheduler_ip}",
+            f"--use_gpu={self.use_gpu}",
+            f"--n_gpus_per_node={self.n_gpus_per_node}",
+        ]
+
+        child_run_config = ScriptRunConfig(
+            source_directory=os.path.join(self.abs_path, "setup"),
+            script="start_worker.py",
+            arguments=args,
+            run_config=run_config,
+        )
+
+        for i in range(workers):
+            child_run = self.run.submit_child(child_run_config)
+            self.workers_list.append(child_run)
+
+    # scale down
+    def scale_down(self, workers=1):
+        for i in range(workers):
+            if self.workers_list:
+                child_run = self.workers_list.pop(0)  # deactive oldest workers
+                child_run.complete()  # complete() will mark the run "Complete", but won't kill the process
+                child_run.cancel()
+            else:
+                print("All scaled workers are removed.")
+
+    # close cluster
+    async def _close(self):
+        if self.status == "closed":
+            return
+        while self.workers_list:
+            child_run = self.workers_list.pop()
+            child_run.complete()
+            child_run.cancel()
+
+        if self.run:
+            self.run.complete()
+            self.run.cancel()
+
+        await super()._close()
+        self.status = "closed"
+        self.__print_message("Scheduler and workers are disconnected.")
+
+    def close(self):
+        """ Close the cluster. All Azure ML Runs corresponding to the scheduler
+        and worker processes will be completed. The Azure ML Compute Target will
+        return to its minimum number of nodes after its idle time before scaledown.
+
+        Example
+        ----------
+        cluster.close()
+        """
+        self.sync(self._close)
```

Comment:
The return from `self.sync` needs to be returned in case it is awaited.

Reply:
Done